### PR TITLE
feat(adapter): clear sensitive caches on user-change

### PR DESCRIPTION
**Pull request by KaiVandivier**
(WIP - awaiting app runtime function) 

Part of a three-part solution to clearing cached data that might be sensitive between users; see [this issue](https://jira.dhis2.org/browse/DHIS2-11759), this [header bar PR](https://github.com/dhis2/ui/pull/740), and this [runtime PR](https://github.com/dhis2/app-runtime/pull/1002).

### CODE REVIEW

File: adapter/src/utils/useVerifyLatestUser.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            const currentUserId = data.user.id
+            if (currentUserId !== latestUserId) {
+                const cachesCleared = await clearSensitiveCaches()
+                localStorage.setItem(LATEST_USER_KEY, currentUserId)
+
+                if (cachesCleared && pwaEnabled) {
+                    // If this is a PWA app, the app-shell cache will need to
+                    // be restored with a page reload
+                    return window.location.reload()
+                }
```
Comment (amcgee) [inline adapter/src/utils/useVerifyLatestUser.js:38]: @KaiVandivier I think this works for now, but I don't think it should be necessary... wouldn't this work as-is if we moved the `useVerifyLastUser` usage up one component level, so it completes before the apps shell fetches anything else?

Comment (KaiVandivier) [reply adapter/src/utils/useVerifyLatestUser.js:38]: Discussed separately, but copied here for others to see: Unfortunately no - if this hook happened above the AppWrapper and below the ServerVersionProvider, the app shell would be missing `api/system/info` in the app shell cache, and would get stuck at the login modal if the app went offline. If the hook happened above ServerVersionProvider, if a user isn't authenticated, then the hook-consuming component wouldn't render the children as-is

**Comment by KaiVandivier** [comment]
Waiting on https://github.com/dhis2/ui/pull/740 to update UI as well in this PR

**Comment by KaiVandivier** [comment]
Ready for re-review, along with 7.x backport: https://github.com/dhis2/app-platform/pull/662

**Comment by varl** [comment]
@KaiVandivier I'm unsure about how to verify this locally.

I tried opening up the PWA example app and logged in with the user `admin`. I recorded and got a success message that things where cached. I verified that items where cached on my client.

I then deleted the authentication cookie to trigger a login without going through the "logout" flow to change the user. I logged in as `system`, and I see cached items with the same timestamps as for the `admin` user.

The logout flow successfully evicts the cache, but not the user-change flow (on expired auth).

Any tips ?

**Comment by KaiVandivier** [comment]
@varl That _is_ the right way to test it, but it works for me 🤔 hmm

**Comment by varl** [comment]
Looks like starting with `--force` refreshed things enough and now it works fine, thanks !

**Review by varl — APPROVED** [review]

**Pull request merged**